pkg/ruler/api.go

```diff
@@ -268,3 +282,274 @@ func (r *Ruler) alerts(w http.ResponseWriter, req *http.Request) {
 		level.Error(logger).Log("msg", "error writing response", "bytesWritten", n, "err", err)
 	}
 }
+
+var (
+	// ErrNoNamespace signals the requested namespace does not exist
+	ErrNoNamespace = errors.New("a namespace must be provided in the url")
+	// ErrNoGroupName signals a group name url parameter was not found
+	ErrNoGroupName = errors.New("a matching group name must be provided in the url")
+	// ErrNoRuleGroups signals the rule group requested does not exist
+	ErrNoRuleGroups = errors.New("no rule groups found")
+	// ErrNoUserID is returned when no user ID is provided
+	ErrNoUserID = errors.New("no id provided")
+	// ErrBadRuleGroup is returned when the provided rule group can not be unmarshalled
+	ErrBadRuleGroup = errors.New("unable to decoded rule group")
+)
+
+// ValidateRuleGroup validates a rulegroup
+func ValidateRuleGroup(g rulefmt.RuleGroup) []error {
+	var errs []error
+	for i, r := range g.Rules {
+		for _, err := range r.Validate() {
+			var ruleName string
+			if r.Alert != "" {
+				ruleName = r.Alert
+			} else {
+				ruleName = r.Record
+			}
+			errs = append(errs, &rulefmt.Error{
+				Group:    g.Name,
+				Rule:     i,
+				RuleName: ruleName,
+				Err:      err,
+			})
+		}
+	}
+
+	return errs
+}
+
+// parseNamespace parses the namespace from the provided set of params, in this
+// api these params are derived from the url path
+func parseNamespace(params map[string]string) (string, error) {
+	namespace, exists := params["namespace"]
+	if !exists {
+		return "", ErrNoNamespace
+	}
+
+	namespace, err := url.PathUnescape(namespace) // namespaces needs to be unescaped if in the URL
```

Comment:
I would instead mention using `router.UseEncodedPath()` in the routes setup, same in `parseGroupName`.

Reply:
👍 `